src/entities/DealTokenSwap.ts

```diff
@@ -204,6 +221,42 @@ export class DealTokenSwap implements IDeal {
     return this.fundingWasInitiated && !this.isExecuted && !this.fundingPeriodHasExpired;
   }
 
+  /**
+   * Gets the DAO based on the connected wallet address
+   */
+  @computedFrom("ethereumService.defaultAccountAddress")
+  public get daoRelatedToWallet(): IDAO{
+    return this.getDao(true);
+  }
+
+  /**
+    * Gets the non-related DAO based on the connected wallet address
+    */
+  @computedFrom("ethereumService.defaultAccountAddress")
+  public get otherDao(): IDAO {
+    return this.getDao(false);
+  }
+
+  /**
+    * Gets the DAO based on the connected wallet address
+    */
+  private getDao(relatedToWallet: boolean) : IDAO {
+    if (this.registrationData.partnerDAO.representatives.some(x => x.address === this.ethereumService.defaultAccountAddress)){
```

Comment:
We can use this.primaryDaoRepresentatives and this.partnerDaoRepresentatives here, and take advantage of Set.has.

Can also use this.isUserProposalLead

Reply:
This is great! I have changed my code to use these new Sets